File: typescript/packages/bench/src/rules/priorityEscalationRule.ts
/** Cardinality fixture: nominal (class, `implements`) implementor of RuleContract. */

import type { RuleContract } from "../contracts/ruleContract";
import type { Dataset } from "../dataset";

export class PriorityEscalationRule implements RuleContract {
    readonly key = "priority-escalation";

    evaluate(data: Dataset): boolean {
        return data.openOrders().length >= 1;
    }
}

File: rust/crates/atelier-bench/src/rules/priority_escalation_rule.rs
//! Rule 17/24 (textual half).

use crate::contracts::rule_contract::RuleContract;
use crate::dataset::Dataset;

pub struct PriorityEscalationRule;

impl RuleContract for PriorityEscalationRule {
    fn evaluate(&self, data: &Dataset) -> bool {
        !data.orders.is_empty()
    }
}
